ci-studio-common-util/cmd/update.go

```diff
@@ -0,0 +1,151 @@
+package cmd
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"runtime"
+
+	"github.com/chef/ci-studio-common/lib"
+	"github.com/juju/fslock"
+	"github.com/mholt/archiver"
+
+	"github.com/spf13/cobra"
+)
+
+var (
+	updateCommand = &cobra.Command{
+		Use:   "update",
+		Short: "Update the ci-studio-common install",
+		Run:   maybeUpdateInstall,
+	}
+
+	phase  int
+	force  bool
+	suffix string
+
+	assetOnDisk = filepath.Join(lib.InstallDirParent(), "ci-studio-common.tar.gz")
+
+	upgradeStatusFile = lib.SettingsPath("upgrade-in-progress")
+)
+
+func init() {
+	rootCmd.AddCommand(updateCommand)
+
+	updateCommand.Flags().IntVar(&phase, "phase", 1, "The phase of the update to complete")
+	updateCommand.Flags().BoolVar(&force, "force", false, "Perform the installation even if no updates are available")
+	updateCommand.Flags().StringVar(&suffix, "suffix", "rc", "The suffix to use when downloading the asset")
+}
+
+func maybeUpdateInstall(cmd *cobra.Command, args []string) {
+	switch phase {
+	case 1:
+		lock := fslock.New(lib.LockPath("upgrade-ci-studio-common"))
+		lockErr := lock.TryLock()
+
+		if lockErr == nil {
+			performPhaseOne()
+		} else {
+			fmt.Println("ci-studio-common upgrade already in progress -- waiting")
+		}
+	case 2:
+		performPhaseTwo()
+	case 3:
+		performPhaseThree()
+	default:
+		fmt.Println("Unsupported phase")
```

Comment:
If you want to send things to stderr you can do:

```
fmt.Fprintf(os.Stderr, "Unsupported phase")
```

We often find it useful to define a little helper in command line tools:

```
func fatal(msg string) {
    fmt.Fprintf(os.Stderr, msg)
    os.Exit(1)
}
```

Or, for tools using logrus: `logrus.Fatal` or  the standard logging library:`log.Fatal`